private/bufpkg/bufprotosource/descriptor.go

```diff
@@ -46,3 +52,16 @@ func (d *descriptor) getLocationByPathKey(pathKey string) Location {
 	}
 	return d.locationStore.getLocationByPathKey(pathKey)
 }
+
+func asDescriptor[T protoreflect.Descriptor](d *descriptor, fullName string, kind string) (T, error) {
+	var zero T
+	desc, err := d.file.resolver.FindDescriptorByName(protoreflect.FullName(fullName))
+	if err != nil {
+		return zero, err
+	}
+	typedDesc, ok := desc.(T)
+	if !ok {
+		return zero, fmt.Errorf("%s is not %s", fullName, kind)
```

Comment:
`"%s is not a %T", fullName, zero` - not sure how kind plays into this one, looks to be unused otherwise